src/Sentry/Internal/ProcessInfo.cs

```diff
@@ -12,43 +14,68 @@ internal class ProcessInfo
         /// <summary>
         /// When the code was initialized.
         /// </summary>
-        internal DateTimeOffset StartupTime { get; set; }
+        internal DateTimeOffset? StartupTime { get; private set; }
 
         /// <summary>
         /// When the device was initialized.
         /// </summary>
-        internal DateTimeOffset BootTime { get; }
+        internal DateTimeOffset? BootTime { get; }
 
-        private SentryOptions _options { get; }
+        private readonly SentryOptions _options;
+        private readonly Func<DateTimeOffset> _findPreciseStartupTime;
 
-        internal ProcessInfo(SentryOptions options)
+        // For testability
+        internal Task PreciseAppStartupTask { get; private set; } = Task.CompletedTask;
+
+        internal ProcessInfo(
+            SentryOptions options,
+            Func<DateTimeOffset>? findPreciseStartupTime = null)
         {
             _options = options;
-            StartupTime = DateTimeOffset.UtcNow;
-            BootTime = DateTimeOffset.UtcNow - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
-        }
+            _findPreciseStartupTime = findPreciseStartupTime ?? GetStartupTime;
+            if (options.DetectStartupTime == StartupTimeDetectionMode.None)
+            {
+                _options.DiagnosticLogger?.LogDebug("Not detecting startup time due to options: {0}",
+                    options.DetectStartupTime);
```

Comment:
Is it bad if you use _options.DetectStartupTime? _options on top of options look different 😅

Reply:
Good point. Will fix